pkg/codegen/go/gen.go

```diff
@@ -4725,11 +4750,14 @@ func generateTypes(
 		hasOutputs = true
 	}
 
+	goInfo := goPackageInfo(pkg.pkg)
 	var goImports []string
 	if hasOutputs {
 		goImports = []string{"context", "reflect"}
 		importsAndAliases["github.com/pulumi/pulumi/sdk/v3/go/pulumi"] = ""
-		importsAndAliases["github.com/pulumi/pulumi/sdk/v3/go/pulumix"] = ""
+		if goInfo.Generics == GenericsSettingSideBySide {
```

Comment:
Isn't this also needed for GenericsOnly?

Reply:
We do however need some test schemas for `generics-only` 🤔